clients/client-java/src/main/java/org/apache/gravitino/client/GenericModelCatalog.java

```diff
@@ -0,0 +1,330 @@
+        restClient.delete(
+            formatModelVersionRequestPath(modelFullIdent)
+                + "/aliases/"
+                + RESTUtils.encodeString(alias),
+            DropResponse.class,
+            Collections.emptyMap(),
+            ErrorHandlers.modelErrorHandler());
+    resp.validate();
+
+    return resp.dropped();
+  }
+
+  /** @return A new builder instance for {@link GenericModelCatalog}. */
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  private void checkModelNamespace(Namespace namespace) {
+    Namespace.check(
+        namespace != null && namespace.length() == 1,
+        "Model namespace must be non-null and only have 1 level, the input namespace is ");
+  }
+
+  private void checkModelNameIdentifier(NameIdentifier ident) {
+    NameIdentifier.check(ident != null, "Model name identifier must be non-null");
+    NameIdentifier.check(
+        StringUtils.isNotBlank(ident.name()), "Model name identifier must have a non-empty name");
+    checkModelNamespace(ident.namespace());
+  }
+
+  private Namespace modelFullNamespace(Namespace modelNs) {
+    return Namespace.of(catalogNamespace().level(0), name(), modelNs.level(0));
+  }
+
+  private NameIdentifier modelFullNameIdentifier(NameIdentifier modelIdent) {
+    return NameIdentifier.of(modelFullNamespace(modelIdent.namespace()), modelIdent.name());
+  }
+
+  @VisibleForTesting
+  static String formatModelRequestPath(Namespace modelNs) {
+    Namespace schemaNs = Namespace.of(modelNs.level(0), modelNs.level(1));
+    return new StringBuilder()
+        .append(formatSchemaRequestPath(schemaNs))
+        .append("/")
+        .append(RESTUtils.encodeString(modelNs.level(2)))
+        .append("/models")
+        .toString();
+  }
+
+  @VisibleForTesting
+  static String formatModelVersionRequestPath(NameIdentifier modelIdent) {
+    return formatModelRequestPath(modelIdent.namespace())
+        + "/"
+        + RESTUtils.encodeString(modelIdent.name());
+  }
+
+  static class Builder extends CatalogDTO.Builder<Builder> {
+
+    private RESTClient restClient;
+
+    private Namespace namespace;
+
+    private Builder() {}
+
+    Builder withNamespace(Namespace namespace) {
+      this.namespace = namespace;
+      return this;
+    }
+
+    Builder withRestClient(RESTClient restClient) {
+      this.restClient = restClient;
+      return this;
+    }
+
+    @Override
+    public GenericModelCatalog build() {
+      Namespace.check(
+          namespace != null && namespace.length() == 1,
+          "Catalog namespace must be non-null and have 1 level, the input namespace is %s",
+          namespace);
+      Preconditions.checkArgument(StringUtils.isNotBlank(name), "name must not be blank");
+      Preconditions.checkArgument(type != null, "type must not be null");
+      Preconditions.checkArgument(StringUtils.isNotBlank(provider), "provider must not be blank");
+      Preconditions.checkArgument(audit != null, "audit must not be null");
```

Comment:
This sounds like a little bit too strict?

Reply:
They are all the necessary variables, it cannot work if one is missing.